lib/improver/tests/blending/weighted_blend/test_WeightedBlendAcrossWholeDimension.py

```diff
@@ -345,59 +508,235 @@ def test_percentiles_different_coordinate_orders(self):
+
+        coord = "forecast_reference_time"
+        plugin = WeightedBlendAcrossWholeDimension(coord, 'weighted_mean')
+        result = plugin.weighted_maximum(self.cube, self.weights3d)
+        expected = np.array([[2.4, 1.2], [1.8, 0.8]])
+
+        self.assertIsInstance(result, iris.cube.Cube)
+        self.assertArrayAlmostEqual(result.data, expected)
+
+    @ManageWarnings(
+        ignored_messages=["Collapsing a non-contiguous coordinate."])
+    def test_without_weights(self):
+        """Test function when a data cube is provided, but no weights cube
+        which should result in equal weightings."""
+
+        coord = "forecast_reference_time"
         plugin = WeightedBlendAcrossWholeDimension(coord, 'weighted_maximum')
-        weights = [0.2, 0.8]
-        result = plugin.process(self.cube, weights)
-        expected_result_array = np.ones((2, 2))*1.6
-        self.assertArrayAlmostEqual(result.data, expected_result_array)
+        result = plugin.weighted_maximum(self.cube, None)
+        expected = np.full((2, 2), 1.)
+
+        self.assertIsInstance(result, iris.cube.Cube)
+        self.assertArrayAlmostEqual(result.data, expected)
 
     @ManageWarnings(
         ignored_messages=["Collapsing a non-contiguous coordinate."])
-    def tests_threshold_splicing_works_weighted_max(self):
-        """Test weighted_max works with a threshold dimension."""
+    def tests_threshold_cube_with_weights_blending_threshold(self):
+        """Test weighted_mean works collapsing a cube with a threshold
+        dimension when the blending is over the threshold coordinate. In this
+        case there is no slicing over the threshold coordinate, so we can test
+        the functionality at this level."""
+        coord = "threshold"
+        plugin = WeightedBlendAcrossWholeDimension(coord, 'weighted_mean')
+        result = plugin.weighted_maximum(self.cube_threshold,
+                                         self.weights_threshold)
+
+        expected_data = np.array([0.16, 0.32, 0.48])
+        expected_array = np.broadcast_to(expected_data, (2, 2, 3)).T
+        self.assertArrayAlmostEqual(result.data, expected_array)
+
+
+class Test_process(Test_weighted_blend):
+
+    """Test the process method."""
+
+    @ManageWarnings(
+        ignored_messages=["Collapsing a non-contiguous coordinate."])
+    def test_basic(self):
+        """Test that the plugin returns an iris.cube.Cube."""
         coord = "time"
-        plugin = WeightedBlendAcrossWholeDimension(coord, 'weighted_maximum')
-        weights = np.array([0.8, 0.2])
-        result = plugin.process(self.cube_threshold, weights)
-        expected_result_array = np.ones((2, 2, 2))*0.4
+        plugin = WeightedBlendAcrossWholeDimension(coord, 'weighted_mean')
+        result = plugin.process(self.cube)
+        self.assertIsInstance(result, Cube)
+
+    def test_fails_coord_not_in_cube(self):
+        """Test it raises CoordinateNotFoundError if coord not in the cube."""
```

Comment:
Can we change to `if blend coord not in the cube` for clarity please.

Reply:
Done.